Source/Core/Core/Core.h

```diff
@@ -74,12 +74,11 @@ void RequestRefreshInfo();
 
 void UpdateTitle();
 
-// waits until all systems are paused and fully idle, and acquires a lock on that state.
-// or, if doLock is false, releases a lock on that state and optionally unpauses.
-// calls must be balanced (once with doLock true, then once with doLock false) but may be recursive.
-// the return value of the first call should be passed in as the second argument of the second call.
-// [NOT THREADSAFE] Host only
-bool PauseAndLock(bool doLock, bool unpauseOnUnlock = true);
+// Run a function on the CPU thread.
+// If called from the CPU thread, the function will be run directly.
+// If called from any other thread, this will first pause and lock the core, then run the function,
+// and finally resume the core.
+void RunOnCPUThread(std::function<void()> function);
```

Comment:
This needs a note that calling it from any thread other than the CPU or host thread is invalid.